### PR TITLE
wifi: move wifi porting to hal

### PR DESCRIPTION
**Pull request by sylvioalves**
all porting related sources moved to hal repository

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

### CODE REVIEW

File: zephyr/adapter/include/wifi/esp_wifi_system.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2020 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (mahavirj) [inline zephyr/adapter/include/wifi/esp_wifi_system.h:6]: Missing header file guards here (`#pragma once`)

File: zephyr/adapter/src/wifi/esp_event.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				}
+			default:
+				break;
+		}
+	}
+}
+
+esp_err_t esp_event_init(void)
+{
+	event_msgq_buffer = k_malloc(sizeof(system_event_t) * 10);
```
Comment (mahavirj) [inline zephyr/adapter/src/wifi/esp_event.c:75]: Can you please checking coding style here once?

File: zephyr/adapter/src/wifi/esp_system_api.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+static unsigned int lock_key;
+
+static const char *TAG = "system_api";
+
+esp_err_t esp_read_mac(uint8_t *mac, esp_mac_type_t type)
+{
+    uint8_t efuse_mac[6] = {0};
+
+    if (mac == NULL) {
+	ESP_LOGE(TAG, "mac address param is NULL");
```
Comment (mahavirj) [inline zephyr/adapter/src/wifi/esp_system_api.c:22]: Please check coding style once for this file

File: zephyr/adapter/include/wifi/esp_networking_priv.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2020 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ESP_NETWORKING_PRIV_H
```
Comment (mahavirj) [inline zephyr/adapter/include/wifi/esp_networking_priv.h:7]: Should we change this to `#pragma once` for consistency?

**Review by mahavirj — CHANGES_REQUESTED** [review]

**Comment by sylvioalves** [comment]
@mahavir, I have pushed the wrong files. That's the reason for all the missing code style.

**Review by mahavirj — APPROVED** [review]

**Pull request merged**